Provide an async example

Niederb opened the pull request:
- Basically a modify the `get_blocks` example such that it takes advantage of async by awaiting multiple futures at the same time. 
- Renamed the example to `get_blocks_async`
- Also integrated it into the CI with a separate compilation and upload step.

Niederb commented on examples/examples/get_blocks_async.rs:
```diff
@@ -0,0 +1,73 @@
+	ac_primitives::PlainTipExtrinsicParams, rpc::JsonrpseeClient, Api, GetChainInfo,
+};
+
+#[cfg(all(feature = "sync-examples", feature = "async-examples",))]
```
The feature shenanigans are a bit annoying... but I couldn't think of a better way to compile the async example (without creating yet another example directory)

haerdib commented on examples/examples/get_blocks_async.rs:
```diff
@@ -0,0 +1,77 @@
+	limitations under the License.
+*/
+
+//! Very simple example that shows how to fetch chain metadata with async.
```
metadata?

Niederb replied:
Yeah I took this from the other example. It might be confusing because metadata has a special meaning in the polkadot eco-system. How about just using `chain information` in both examples?

haerdib replied:
Works  for me 👍

haerdib commented on examples/examples/get_blocks.rs:
```diff
@@ -27,6 +28,7 @@ use substrate_api_client::{
 #[tokio::main]
 async fn main() {
```
What do you think: Do we still need the sync variant of get_blocks? We have different sync examples and the difference between sync and async is not that big.

Niederb replied:
Initially I thought it would be nice to show that the async version is faster. Unfortunately that's not really the case. The sync has the small advantage of outputting something directly after each getter, so if something fails later on there is at least the output of the previous stuff.

Niederb commented:
- Fixed the metadata comment as discussed
- Removed the sync version of `get_blocks`
  - Integrated the async version instead in the CI. This needs a separate compilation and update step.
  - Added the synchronous subscription part to the async example such that is still there for reference and testing.

haerdib reviewed (approved):
Absolutely awesome, thanks a lot! One last wish: Could you update the PR description? 😇
